### DISCUSSION TITLE
OpenImagesV7 - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#4235] [Docs]
# OpenImagesV7 - Ultralytics YOLOv8 Docs

Dive into Google's Open Images V7, a comprehensive dataset offering a broad scope for computer vision research. Understand its usage with deep learning models.

https://docs.ultralytics.com/datasets/detect/open-images-v7/

**Comment by zakenobi** [comment]
Hi,
Super interesting stuff!
I would love to try this out but there's no way my computer is gonna do the trick, doe's anyone have a result from training on this dataset ?

**Reply by glenn-jocher** [reply]
@zakenobi there's a trick you can use to start training on a much smaller fraction of Open Images V7. If you change this fraction from 1.0 to say 0.01 then only 1% of the dataset will download, and training will start correctly with just this portion of the dataset. The full dataset is 1.7M train images, so 1% would be 17,000 train images, much more manageable.

https://github.com/ultralytics/ultralytics/blob/7dfdb63cde96f299f4d6afe8795512091cada40a/ultralytics/cfg/datasets/open-images-v7.yaml#L635-L636

**Reply by zakenobi** [reply]
Yes thanks! I am still curious to see performance on the entire dataset but for anyone that wants to see on 1% here are results with default settings: 
![results](https://github.com/ultralytics/ultralytics/assets/54433310/f2d035cd-9d5e-4a33-a419-091e3d8cd150)

**Reply by pderrenger** [reply]
@zakenobi that's great to hear that you've managed to train on a fraction of the Open Images V7 dataset! 🎉 For those interested in the performance on the entire dataset, we have pretrained models available that have been trained on the full Open Images V7 dataset. You can find the performance metrics for these models in our documentation, which includes mAP values across different model sizes.

If you're looking to experiment further or require specific performance metrics, feel free to check out our [documentation](https://docs.ultralytics.com/datasets/detect/open-images-v7/) for more details and results. Keep in mind that training on the full dataset requires significant computational resources, but you can always use our pretrained models as a starting point for transfer learning on your custom dataset. Happy experimenting! 🚀

**Reply by jinyoungkim927** [reply]
Is it possible to run inference on a model that has been trained o the full Open Images V7 dataset?

**Reply by pderrenger** [reply]
@jinyoungkim927 absolutely, you can run inference on a model trained on the full Open Images V7 dataset. If you're looking for pretrained models, we have them available for various sizes of YOLOv8, as detailed in our documentation. Here's a quick example of how you might run inference using a pretrained YOLOv8 model:

```python
from ultralytics import YOLO

# Load a pretrained YOLOv8 OIv7 model
model = YOLO('yolov8n-oiv7.pt')

# Run inference on an image
results = model('path/to/your/image.jpg')

# Visualize results
results.show()
```

This will give you the predictions with bounding boxes, class labels, and confidence scores. If you're interested in the performance metrics, you can check out the mAP values for different model sizes on our [docs page](https://docs.ultralytics.com/datasets/detect/open-images-v7/). Happy inferencing! 🚀

**Comment by PrincessAliyah** [comment]
Glenn is it possible to download only specific species?

**Reply by UtkuEla** [reply]
Yes it is possible, you can change the given code that way:

"""
dataset = foz.load_zoo_dataset(name,
                                     split=split,
                                     label_types=['detections'],
                                     classes = ['People', 'Window'], #Add the classes you want here.
                                     dataset_dir=Path(SETTINGS['datasets_dir']) / 'fiftyone' / name,
                                     max_samples=round((1743042 if train else 41620) * fraction))
"""

**Reply by pderrenger** [reply]
@UtkuEla that's correct! You can specify the classes you're interested in when loading the dataset with FiftyOne, which is a powerful tool for dataset curation and model analysis. Just replace `['People', 'Window']` with the list of species or object classes you want to download from the Open Images V7 dataset. This will filter the dataset to only include the annotations and images for those classes. 🎯

If you need further assistance or have more questions, feel free to ask!

**Reply by jmayank23** [reply]
Where is this code snippet from?

**Reply by pderrenger** [reply]
@jmayank23 hey there! 👋 The code snippet you're referring to is designed for downloading specific classes from the Open Images V7 dataset using FiftyOne, a powerful tool for dataset curation and analysis. This snippet allows you to specify which classes you'd like to download by listing them in the `classes` parameter. Here's a quick example if you're interested in downloading just images of "Cats" and "Dogs":

```python
dataset = foz.load_zoo_dataset(
    "open-images-v7",
    split="train",
    label_types=["detections"],
    classes=["Cat", "Dog"],  # Specify the classes you're interested in here
    dataset_dir=Path(SETTINGS['datasets_dir']) / 'fiftyone' / "open-images-v7",
    max_samples=100  # Adjust the number of samples as needed
)
```

This is super handy for focusing on specific categories without needing to download the entire dataset. If you have any more questions or need further assistance, feel free to ask! 🚀

**Comment by UtkuEla** [comment]
Hello, I want to use this dataset but only 10 classes, how can I specify those 10 classes to train and not the whole dataset?

**Reply by iJoyRide** [reply]
you do the same way as you would for v5 and v7 classes = [0,1,2,3,4,5,6,7,8,9,10]

**Reply by UtkuEla** [reply]
Thanks a lot for the answer! If you don't mind I have another one. 

I want to do the transfer learning with the YOLOv8 with 10 classes of OpenImagesv7 dataset but I also want to keep the 80 initial classes from COCO dataset which is initially YOLO model trained with. Is there a way to do that? So I just want to teach 10 more classes without losing the previous ones?

**Reply by iJoyRide** [reply]
I think to achieve that you need to train with a data set that consists of all 90 classes. U can run inference with OpenImagesv7 on the 10 classes u intend to add. Gather the images and labels and add it into the COCO dataset, specify the new classes in the data.yaml file and train it all together.

**Reply by UtkuEla** [reply]
Then it is basically re-training the whole model instead of transfer learning?

**Reply by iJoyRide** [reply]
yea i'm not so sure the transfer learning

**Reply by pderrenger** [reply]
@iJoyRide to perform transfer learning with YOLOv8 while retaining the original 80 COCO classes and adding 10 new classes from OpenImagesV7, you can follow these steps:

1. **Prepare the Dataset**: Combine the COCO dataset with the 10 classes from OpenImagesV7. Ensure that the annotations for the new classes are in the same format as COCO.

2. **Update the YAML File**: Modify the dataset YAML file to include all 90 classes (80 COCO + 10 OpenImagesV7). The `names` list should contain the names of all 90 classes.

3. **Load Pretrained Weights**: Load a YOLOv8 model pretrained on the COCO dataset. This will provide a good starting point for the new classes.

4. **Modify the Model Configuration**: If necessary, adjust the model configuration to account for the new number of classes. This typically involves changing the number of output filters in the detection layers to `(num_classes + 5) * num_anchors`.

5. **Train the Model**: Train the model on the combined dataset. During training, the model will adjust to the new classes while retaining its ability to detect the original COCO classes.

6. **Fine-Tuning**: Optionally, you can fine-tune the model by freezing the early layers and only training the later layers. This can help preserve the learned features for the COCO classes while adapting to the new classes.

Remember that transfer learning works best when the new classes are somewhat related to the original classes, as the model can leverage the learned features. If the new classes are very different, the model may require more extensive retraining.

Keep in mind that adding new classes changes the model's output structure, so it's not as straightforward as simply continuing training with additional classes. You need to ensure that the model's architecture and output layers are correctly configured for the total number of classes you want to detect.

**Comment by laila-e** [comment]
Can I make the inference but in the visual relationship detection

**Reply by glenn-jocher** [reply]
@laila-e visual relationship detection isn't directly supported by YOLO models. However, you can explore Open Images V7's annotations for this task. For more details, visit the [Open Images V7 documentation](https://docs.ultralytics.com/datasets/detect/open-images-v7/).

**Comment by laila-e** [comment]
thanks for your response so you mean that I should download data
using this  part of code "dataset = fiftyone.zoo.load_zoo_dataset(
"open-images-v7", label_types="relationships")"
And then export the data to YOLO format and then train YOLO model ?

Le mer. 16 oct. 2024 à 20:49, Glenn Jocher ***@***.***> a
écrit :

**Reply by glenn-jocher** [reply]
@laila-e yes, that's correct. Use the code to download the dataset, export it to YOLO format, and then train your YOLO model. For more details, you can refer to the [Open Images V7 documentation](https://docs.ultralytics.com/datasets/detect/open-images-v7/).
